app/src/main/java/com/mobile/tandil/javabaseproject/mvp/view/ParkingView.java

```diff
@@ -29,4 +34,19 @@ public void showDialogFragment(ListenerDialogFragment listenerDialogFragment) {
     public void showNumberOfParking(String numberOfParking) {
         binding.textViewMainActivity.setText(numberOfParking);
     }
+
+    @Override
+    public void showReservationActivity() {
+        Context context = getContext();
+        Activity activity = getActivity();
+        if (context != null && activity != null) {
+            Intent i = new Intent(context, ReservationActivity.class);
+            activity.startActivity(i);
+        }
+    }
+
+    @Override
+    public void showToastEmptyAvailableSpaces() {
+        Toast.makeText(getActivity(), R.string.reservation_button_main_activity, Toast.LENGTH_SHORT).show();
```

Comment:
What happens if the activity is null?

Reply:
It could crash the app? I already added a if sentece